Sources/SwiftSyntaxMacrosGenericTestSupport/Assertions.swift

```diff
@@ -0,0 +1,607 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=6.0)
+import SwiftBasicFormat
+public import SwiftDiagnostics
+@_spi(FixItApplier) import SwiftIDEUtils
+import SwiftParser
+import SwiftParserDiagnostics
+public import SwiftSyntax
+public import SwiftSyntaxMacroExpansion
+private import SwiftSyntaxMacros
+private import _SwiftSyntaxGenericTestSupport
+#else
+import SwiftBasicFormat
+import SwiftDiagnostics
+@_spi(FixItApplier) import SwiftIDEUtils
+import SwiftParser
+import SwiftParserDiagnostics
+import SwiftSyntax
+import SwiftSyntaxMacroExpansion
+import SwiftSyntaxMacros
+import _SwiftSyntaxGenericTestSupport
+#endif
+
+/// Defines the location at which the a test failure should be anchored. This is typically the location where the
+/// assertion function is called.
+public struct TestFailureLocation {
+  @_spi(XCTestFailureLocation) public let staticFileID: StaticString
```

Comment:
These symbols are _only_ used by XCTest. The ones used by a caller using swift-testing use the normalized types (`String` and `Int`) instead of the types used by `XCTFail()` (`StaticString` and `UInt`.) This was an intentional decision. :)